package/MDAnalysis/analysis/encore/clustering/Cluster.py

```diff
@@ -0,0 +1,211 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Ensemble representation --- :mod:`MDAnalysis.analysis.ensemble.ensemble`
+=====================================================================
+
+The module contains the Cluster and ClusterCollection classes which are
+designed to store results from clustering algorithms.
+
+:Author: Matteo Tiberti, Wouter Boomsma, Tone Bengtsen
+:Year: 2015--2016
+:Copyright: GNU Public License v3
+:Mantainer: Matteo Tiberti <matteo.tiberti@gmail.com>, mtiberti on github
+
+.. versionadded:: 0.14.0
+
+"""
+
+import numpy as np
+
+
+class Cluster(object):
+    """
+    Generic Cluster class for clusters with centroids.
+
+    Attributes
+    ----------
+
+    id : int
+        Cluster ID number. Useful for the ClustersCollection class
+
+    metadata : iterable
+        dict of lists, containing metadata for the cluster elements. The
+        iterable must return the same number of elements as those that
+        belong to the cluster.
+
+    size : int
+        number of elements.
+
+    centroid : element object
+        cluster centroid.
+
+    elements : numpy.array
+        array containing the cluster elements.
+   """
+
+    def __init__(self, elem_list=None, centroid=None, idn=None, metadata=None):
+        """Class constructor. If elem_list is None, an empty cluster is created
+            and the remaining arguments ignored.
+
+        Parameters
+        ----------
+
+        elem_list : numpy.array or None
+            numpy array of cluster elements. if None, the cluster will be
+            initialized as empty.
+
+        centroid : None or element object
+            centroid object
+
+        idn : int
+            cluster ID
+
+        metadata : {str:iterable, ...}
+            metadata, one value for each cluster element. The iterable
+            must have the same length as the elements array.
+
+    """
+
+        self.id = idn
+
+        if elem_list is None:
+            self.size = 0
+            self.elements = np.array([])
+            self.centroid = None
+            self.metadata = {}
+            return
+
+        self.metadata = {}
+        self.elements = elem_list
+        if centroid not in self.elements:
+            raise LookupError
+
+        self.centroid = centroid
+        self.size = self.elements.shape[0]
+        if metadata:
+            for k, v in metadata.iteritems():
```

Comment:
Since you have the nice `add_metadata` function you might want to use

```python
for name, data in six.iteritems(metadata):
    self.add_metadata(name, data)
```